Role descriptions

Cadair opened the pull request:
This also includes a bunch of refactoring and internal changes to lay this out properly, but I would like everyone to review the [roles.rst](https://github.com/Cadair/sunpy.org/blob/841e4e7bc3b579534a9fc2270f567eaad871a5b3/project/roles.rst) file.

The built version is [here](https://deploy-preview-202--sunpy.netlify.com/project/roles.html), the navigation bar is incorrect in the preview because this PR needs an accompanying PR to the theme repo (https://github.com/sunpy/sunpy-sphinx-theme/pull/96)

@sunpy/sunpy-board

ehsteve commented:
My main comment is that I don't think the nitty gritty description and responsibilities of the roles need to be on the website proper. They should be described at a high level on the website with the current person's name (and photo!) and a link to the detailed description (which I see as more of an internal but public document).

Cadair commented:
I very much disagree.

Firstly, I think if we are going to put peoples names (and faces) on the website with a title attached we should also have a description of what they do somewhere so people can read it.

Secondly, all our documentation is public somewhere, the website seems like a sensible place to keep this, given the website and the SEPs are the only place we really keep this information at the moment.

Thirdly, you also PRed the first role description to the website for the subpackage maintainers, which I have incorporated into this new page.

Lastly, Astropy also have their role descriptions on their main team page.

Cadair commented:
Also thinking about this a bit more, I think that the sunpy.org website is exactly the place we should be documenting the project. In my mind that includes, the people, the roles, the organisation structure etc.

Obviously I am not suggesting we put this all at the top level and throw it in peoples faces, the top of the website nav tree needs to focus on users and people researching sunpy. However, I think that a few layers of the tree down where people are only going to find it if they are looking or curious, it makes perfect sense for this to be our project docs site.

wtbarnes commented:
On the `people/community_roles` page, it lists "Webmaster" as unfilled, but on the `project/roles` it lists Nabil as the Webmaster.

Cadair commented:
Yeah that's a todo to fix.

nabobalis commented on project/roles.rst:
```diff
@@ -0,0 +1,172 @@
+
+Webmaster
+---------
+**Nabil Freij**
```
```suggestion
**Unfilled**
```
As per comments.

nabobalis commented on project/index.rst:
```diff
@@ -0,0 +1,325 @@
+                <p><a href="https://github.com/sunpy/sunpy/wiki/Minutes-of-SunPy-Board-Meeting-20140416">Board Meeting 2014-04-16</a></p>
+                <p><a href="https://github.com/sunpy/sunpy/wiki/Minutes-of-SunPy-Board-Meeting-20140407">Board Meeting 2014-04-07</a></p>
+              </details>
+
```
I think this should be moved to another page or elsewhere.

Cadair replied:
I agree, but I don't have a good place for it right now so...

nabobalis commented on conf.py:
```diff
@@ -50,6 +50,8 @@
 default_role = "obj"
 
+templates_path += ["_templates"]
+
```
```suggestion
```

nabobalis commented on conf.py:
```diff
@@ -50,6 +50,8 @@
 
 default_role = "obj"
 
```
```suggestion
```

nabobalis commented on Makefile:
```diff
@@ -18,3 +18,7 @@ help:
 %: Makefile
 	@$(SPHINXBUILD) -M $@ "$(SOURCEDIR)" "$(BUILDDIR)" $(SPHINXOPTS) $(O)
+
+
```
```suggestion
```

nabobalis commented:
Will this fix #196 or #192

Cadair commented on project/roles.rst:
```diff
@@ -0,0 +1,172 @@
+
+
+**Commitment Level**: The lead developer + deputy roles combined need a large time commitment.
+Between running the weekly meetings and ensuring that PRs are reviewed and issues triaged, the shared time commitment of the two roles is probably between 10 and 20 hours a week.
```
```suggestion

```

Cadair commented on project/roles.rst:
```diff
@@ -0,0 +1,172 @@
+comprises of the following main tasks:
+
+-  Ensuring that PRs which need backporting into release branches are backported.
+-  Maintaining the CI on release branches.
```
```suggestion
-  Coordinating with the CI maintainer to maintain the CI on release branches.
```

Cadair commented:
Thanks @ehsteve !

nabobalis commented:
> Will this fix #196 or #192

Will it?!

Cadair commented:
No.

nabobalis commented:
> No.

:(

dpshelio commented on exts/cards.py:
```diff
@@ -74,42 +66,39 @@ class Card(Directive):
+        if "title" in self.options:
+            title = self.options.get("title")
+        else:
+            title = ""
```
Are these `options` an object with `get` that doesn't allow a default output?
```python
title = self.options.get("title", "")
```

Cadair replied:
I am honestly not sure, I will investigate this in a follow-on PR when I attack this a little more.

nabobalis replied:
Yeah, this entire code needs tidying up but not here.

dpshelio commented on project/affiliated.rst:
```diff
@@ -0,0 +1,49 @@
+
+An affiliated package is a Python package related to Solar Physics that is not part of the SunPy core library, but can be included in the future as part of the SunPy project’s community.
+
+As a community-driven project SunPy thus encourages contributions from a diverse group of people on building such software that has the potential to be a future **Affiliated SunPy Package.**
```
```suggestion
As a community-driven project SunPy thus encourages contributions from a diverse group of people on building such software that has the potential to be a future **Affiliated SunPy Package**.
```

dpshelio commented on project/affiliated.rst:
```diff
@@ -0,0 +1,49 @@
+*  The developers of an affiliated package should engage with the rest of the SunPy community to encourage knowledge and code sharing within
+   the community.
+
+Please look at `this SEP`_ to read about our policies surrounding affiliated packages.
```
```suggestion
Please look at `SEP-4`_ to read about our policies surrounding affiliated packages.
```
`this` link, `here`, etc... are not descriptive links

dpshelio commented on project/affiliated.rst:
```diff
@@ -0,0 +1,49 @@
+   the community.
+
+Please look at `this SEP`_ to read about our policies surrounding affiliated packages.
+Please send an email to the `mailing list`_ to start a dialogue.
```
```suggestion
Please send an email to `our mailing list`_ to start a dialogue.
```

dpshelio commented on project/affiliated.rst:
```diff
@@ -0,0 +1,49 @@
+Please look at `this SEP`_ to read about our policies surrounding affiliated packages.
+Please send an email to the `mailing list`_ to start a dialogue.
+
+.. _this SEP: https://github.com/sunpy/sunpy-SEP/blob/master/SEP-0004.md
```
```suggestion
.. _SEP-4: https://github.com/sunpy/sunpy-SEP/blob/master/SEP-0004.md
```

dpshelio commented on project/affiliated.rst:
```diff
@@ -0,0 +1,49 @@
+Please send an email to the `mailing list`_ to start a dialogue.
+
+.. _this SEP: https://github.com/sunpy/sunpy-SEP/blob/master/SEP-0004.md
+.. _mailing list: https://groups.google.com/forum/#!forum/sunpy
```
```suggestion
.. _our mailing list: https://groups.google.com/forum/#!forum/sunpy
```

dpshelio commented on project/affiliated.rst:
```diff
@@ -0,0 +1,49 @@
+   * - `radiospectra <https://github.com/sunpy/radiospectra>`_
+     -  This package aims to provide support for some type of radiospectra on solar physics
+     - `radiospectra docs <https://docs.sunpy.org/projects/radiospectra>`_
+     - `David Pérez-Suárez`_
```
:joy:

nabobalis replied:
You ok?

dpshelio commented on project/index.rst:
```diff
@@ -0,0 +1,325 @@
+The **SunPy project** is the organization which maintains the SunPy library and sponsored affiliated packages.
+Its primary goal is to *facilitate and promote the use and development of a community-led, free and open-source solar data-analysis software based on the scientific Python environment*.
+
+The organization is described in `this document`_, and consists of a board of directors, a lead developer appointed by the board, community roles and the sunpy and affiliated package developers.
```
```suggestion
The organization is described in `the SEP-2`_, and consists of a board of directors, a lead developer appointed by the board, community roles and the sunpy and affiliated package developers.
```
Otherwise, `this document` may be the current one I'm reading.

dpshelio commented on project/roles.rst:
```diff
@@ -0,0 +1,169 @@
+-  Updating the tox configurations.
+-  Investigating new CI services and recommending which ones could be useful.
+
+**Commitment Level Required**: Low to Med (3-5 hours per week). This role requires keeping an eye on the core repo and other repos’s PRs to spot issues with the CI systems and builds which needs attention.
```
```suggestion
**Commitment Level Required**: Low to Med (3-5 hours per week). This role requires keeping an eye on the core repo and other repos’ PRs to spot issues with the CI systems and builds which needs attention.
```
I'm not sure how it works in this case... `repos` is the short for the plural, therefore I believe the `s` after the aphostrophe is not needed.

mbobra commented:
Really nice write-up! 

Perhaps we can drop the qualifications of the commitment level (e.g. Low/Medium/High) and simply state the hours? The Release Manager commitment level says "Low to Med (3-5 hours per week)" whereas the CI Manager commitment level says "Low (3-5 hours/week)". 5 hours a week is 8% of one's time -- perhaps not low for some.

Cadair commented:
@mbobra I have updated it to remove those, I agree they are subjective, and even relative to one another they add little value. I have also changed the wording to make it clear the hours are an estimate.
